src/test/java/com/alipay/oceanbase/rpc/ObTableClientTest.java

```diff
@@ -2689,4 +2691,571 @@ public void testAggregationWithIllegalColumn() throws Exception {
+
+                // test insertup exist, update again
+                theResult = client.insertOrUpdate("test_auto_increment_rowkey")
+                        .setRowKey(colVal("c1", 103), colVal("c2", 1))
+                        .addMutateRow(row(colVal("c3", 50))).execute();
+
+                tableQuery.select("c1", "c2", "c3");
+                filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 103);
+                tableQuery.setFilter(filter);
+                result = tableQuery.execute();
+                result.next();
+                value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals(50, value.get("c3"));
+
+                // test increment not exist, insert
+                value = client.increment("test_auto_increment_rowkey", new Object[] {0, 1},
+                        new String[] { "c3" }, new Object[] { 6 }, true);
+
+                tableQuery.select("c1", "c2", "c3");
+                filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 104);
+                tableQuery.setFilter(filter);
+                result = tableQuery.execute();
+                result.next();
+                value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals(6, value.get("c3"));
+
+                // test increment exist, increment
+                value = client.increment("test_auto_increment_rowkey", new Object[] {104, 1},
+                        new String[] { "c3" }, new Object[] { 6 }, true);
+
+                tableQuery.select("c1", "c2", "c3");
+                filter = new ObTableValueFilter(ObCompareOp.EQ, "c1", 104);
+                tableQuery.setFilter(filter);
+                result = tableQuery.execute();
+                result.next();
+                value = result.getRow();
+                Assert.assertEquals(1, value.get("c2"));
+                Assert.assertEquals(12, value.get("c3"));
+
+                // test illegal increment on auto increment column
+                try {
+                    value = client.increment("test_auto_increment_rowkey", new Object[] {104, 1},
+                            new String[] { "c1" }, new Object[] { 1 }, true);
+                } catch (ObTableException e) {
+                    assertNotNull(e);
+                    assertEquals(ResultCodes.OB_NOT_SUPPORTED.errorCode, e.getErrorCode());
+                }
+            } finally { // use sql to drop table
+                statement.execute("drop table test_auto_increment_rowkey");
+            }
+        }
+    }
+
+    @Test
+    // Test auto increment on rowkey for append
+    public void testAutoIncrementRowkeyAppend() throws Exception {
+        if (ObGlobal.OB_VERSION <= 0) {
+            // ob version is invalid
+            Assert.assertTrue(false);
+        } else if (ObGlobal.OB_VERSION == 4) {
+            // todo: auto increment only support in 4.x currently
```

Comment:
Panic when the version is not equal to 4, and leave todo there.

Reply:
fixed